server/covmanager/views.py

```diff
@@ -588,30 +588,54 @@ class CollectionViewSet(mixins.CreateModelMixin,
     ]
 
 
-class ReportViewSet(mixins.ListModelMixin,
-                    mixins.RetrieveModelMixin,
-                    viewsets.GenericViewSet):
+class ReportFilterBackend(filters.BaseFilterBackend):
     """
-    API endpoint that allows viewing aggregated Collections (prefiltered)
+    Accepts broad filtering by q parameter to search multiple fields
     """
-    authentication_classes = (TokenAuthentication, SessionAuthentication)
+    def filter_queryset(self, request, queryset, view):
+        """
+        Return a filtered queryset.
+        """
+        # Return early on empty queryset
+        if not queryset:
+            return queryset
 
-    report_filters = getattr(settings, 'COV_REPORTS_FILTER', ["Report"])
-    report_filters = [Q(description__contains=v) for v in report_filters]
-    filter = report_filters.pop()
+        can_see_unpublished = False
+        if request.user and request.user.is_authenticated:
+            can_see_unpublished = request.user.has_perm('crashmanager.view_crashmanager')
```

Comment:
We should create a separate permission for this in the future.

Reply:
Yes, I left a comment for that in another place where I used the crashmanager permission.